### PR TITLE
Add FFX FSR as a dynamic resolution setting

### PR DESCRIPTION
**Pull request by pixelflinger**

### CODE REVIEW

File: filament/backend/src/opengl/OpenGLProgram.cpp [hunk 1]
```diff
@@ -92,7 +92,28 @@ float u16tofp32(highp uint v) {
...
+    } else if (exponent > -15) {
+        exponent += 15;
+        mantissa >>= 13;
+        f16 = sign | uint(exponent << 10) | mantissa;
+    } else {
+        f16 = sign;
+    }
+    return f16;
+}
+highp uint packHalf2x16(vec2 v) {
```
Comment (romainguy) [inline filament/backend/src/opengl/OpenGLProgram.cpp:112]: Can we use a built-in when it's available?

File: filament/src/PostProcessManager.cpp [hunk 2]
```diff
@@ -226,6 +227,8 @@ static const MaterialInfo sMaterialList[] = {
         { "separableGaussianBlur",      MATERIAL(SEPARABLEGAUSSIANBLUR) },
         { "taa",                        MATERIAL(TAA) },
         { "vsmMipmap",                  MATERIAL(VSMMIPMAP) },
+        { "fsr_easu",                  MATERIAL(FSR_EASU) },
```
Comment (romainguy) [inline filament/src/PostProcessManager.cpp:230]: Nit: alignment on `MATERIAL` (same the line below)

**Review by romainguy — APPROVED** [review]

File: filament/backend/src/opengl/OpenGLProgram.cpp (hunk 1 above)
Comment (pixelflinger) [reply filament/backend/src/opengl/OpenGLProgram.cpp:112]: Well, it should inject that code only on GL 4.1 when the extension is not available.

Comment (romainguy) [reply filament/backend/src/opengl/OpenGLProgram.cpp:112]: Ah right, sorry the diff on GitHub wasn't showing enough context.

**Pull request merged**